# ETFs = Needs cleaning and checking/removing of Duplicates, renaming, stitching, etc. We deal with this later? NOPE! 

# Sector/Industry ETFs > Indices

# For a low-budget systematic trader, this ETF universe is actually powerful because it lets you model:

Global country rotation

Sector rotation

Industry rotation

Risk-on / risk-off

Bear market leadership

Relative strength

Intermarket relationships

# without needing expensive survivorship-bias-free constituent databases for every stock universe. In practice, many institutional top-down models are built from a universe of only ~100-300 highly liquid ETFs rather than thousands of individual equities.

# Avoid un-vetted Thematic ETFs.
#### Cos Thematic ETFs notoriously don't always capture the underlying! BOTZ underperforming QQQ for last decade, Crypto ETF underperforming BTC due to it's own corporate issues, etc.
#### BUT Stocks are better than Thematic ETFs (risks below) AND Leading Stocks lead Themes and Thematic ETFs. Don't waste your time on Thematic ETFs unless vetted! LATER
https://www.reddit.com/r/ETFs/comments/1twjlel/thematic_etfs/

#### Let's just get the major ETFs for now!

# Adding ETPs
We now only have a list of stocks. But we might also want to trade SPY, TLT or VXX. I am only interested in a selection of ETPs and indices; these were found by taking the most liquid ETF in its category at [etfdb.com](etfdb.com).

* Size ETFs: "SPY", "IJH", "IJR", 
    * Large-cap, Mid-cap, Small-cap, 
* Bond ETFs: "BIL", "SHY", "IEF", "TLT", "JPST", "VCSH", "VCIT", "VCLT" 
    * risk-free (1-3M, 1-3Y, 7-10Y, 20+Y), corporate (Overnight, 1-5Y, 5-10Y, 10-25Y)
* Regions: "EEM", "IXUS", + SPY
* Region/Larger countries ETFs: "FXI", "EEM", "VGK", "EWY", "EWZ", "EWJ", "INDA", "EWW" + SPY
    * Based on the Finviz map the largest region ETFs.
* Country ETFs: 'EWA', 'EWO', 'EWK', 'EWC', 'EWQ', 'EWG', 'EWH', 'EWI', 'EWJ', 'EWM', 'EWW', 'EWN', 'EWS', 'EWP', 'EWD', 'EWL', 'EWU' + SPY
    * The iShares ETFs that were released before 2000 and thus have a very long history of data. Often the most liquid.
* Sector ETFs: "XLE", "XLU", "XLK", "XLB", "XLP", "XLY", "XLC", "XLV", "XLF", "XLRE"
    * The sectors that Finviz also uses.
* Commodity ETFs "GLD", "SLV"
    * Gold, silver
* Volatility ETPs: "VXX", "UVIX", "SVIX", "SVXY"
    * Long, Long 2X, Short 1x, Short 0.5x (1x before and including 2018-02-27)

*Note: ETPs is an umbrella term for ETFs, ETVs and ETNs. However most investors just refer to them as 'ETFs'.*

In [1]:
from massive.rest import RESTClient
from datetime import date
from tickers import get_tickers
from times import first_trading_date_after_equal, last_trading_date_before_equal
import os
import pandas as pd
import numpy as np
import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)
from dotenv import load_dotenv 

# Now load variables and define constants
load_dotenv() # to load variables from .env

ACCESS_KEY_ID = os.environ.get("ACCESS_KEY_ID")
if not ACCESS_KEY_ID:
    raise ValueError("No API key found. Please set ACCESS_KEY_ID in your .env file.")

API_KEY = os.environ.get("MASSIVE_API_KEY")
if not API_KEY:
    raise ValueError("No API key found. Please set MASSIVE_API_KEY in your .env file.")

client = RESTClient(api_key=API_KEY)

POLYGON_DATA_PATH = "../data/polygon/"
START_DATE = first_trading_date_after_equal(date(2016, 6, 6))
END_DATE = last_trading_date_before_equal(date(2026, 6, 3))

The ETFs can be added here manually.

In [2]:
# Broad Market / Size ETFs
size_ETFs = [
    "SPY", "IVV", "VOO", # S&P 500
    "QQQ",               # Nasdaq 100
    "DIA",               # Dow Jones
    "MDY", "IJH", "VO",  # Mid-Cap
    "IWM", "IJR", "VB"   # Small-Cap
]

# Factor & Style ETFs
factor_ETFs = [
    "VTV", "IVE", "VLUE", # Value
    "VUG", "IVW",         # Growth
    "MTUM",               # Momentum
    "QUAL",               # Quality
    "USMV", "SPLV",       # Low Volatility
    "SCHD", "VIG", "VYM"  # Dividend Growth/Yield
]

# Fixed Income / Bond ETFs
bond_ETFs = [
    "BIL", "SHV", "SHY",  # Ultra-Short / Short Treasury
    "IEI", "IEF",         # Intermediate Treasury
    "TLT", "EDV",         # Long-Term Treasury
    "AGG", "BND",         # Total Aggregate Bond
    "LQD", "JPST", "VCSH", "VCIT", "VCLT", # Inv Grade Corporate # risk-free (1-3M, 1-3Y, 7-10Y, 20+Y), corporate (Overnight, 1-5Y, 5-10Y, 10-25Y)
    "HYG", "JNK",         # High Yield Corporate
    "MBB",                # Mortgage-Backed
    "MUB"                 # Municipal
]

# Broad International / Region ETFs
region_ETFs = [
    "VEA", "EFA",         # Developed ex-US
    "VWO", "EEM",         # Emerging Markets
    "VXUS", "IXUS"        # Total Int ex-US
]

# Sub-Region ETFs
subregion_ETFs = [
    "VGK", "EZU",         # Europe
    "VPL", "AAXJ",        # Asia Pacific
    "ILF"                 # Latin America
]

# Single Country ETFs
country_ETFs = [
    "EWC", "EWU", "EWG", "EWQ", "EWL", "EWI", "EWP", "EWD", 
    "EWJ", "EWA", "EWY", "EWT", "INDA", "FXI", "MCHI", 
    "EWZ", "EWW", "EZA", "KSA", "EWO", "EWK", "EWH", "EWM", "EWN", "EWS", 
]

# Leveraged ETFs (Bull - 2x and 3x)
leveraged_ETFs = [
    "UPRO", "SSO",        # 3x and 2x S&P 500
    "TQQQ", "QLD",        # 3x and 2x Nasdaq 100
    "UDOW",               # 3x Dow Jones
    "TNA",  "UWM",        # 3x and 2x Russell 2000
    "SOXL",               # 3x Semiconductors
    "TECL",               # 3x Technology Sector
    "FAS",                # 3x Financials
    "TMF",                # 3x Long-Term Treasury (TLT equivalent)
    "UCO"                 # 2x Crude Oil
]

# Inverse & Leveraged Inverse ETFs (Bear - -1x, -2x, and -3x)
inverse_ETFs = [
    "SPXU", "SDS", "SH",  # -3x, -2x, and -1x S&P 500
    "SQQQ", "QID", "PSQ", # -3x, -2x, and -1x Nasdaq 100
    "SDOW",               # -3x Dow Jones
    "TZA",  "RWM",        # -3x and -1x Russell 2000
    "SOXS",               # -3x Semiconductors
    "TECS",               # -3x Technology Sector
    "FAZ",                # -3x Financials
    "TMV"                 # -3x Long-Term Treasury
]

volatility_ETFs = [
    "VXX", "UVIX", "SVIX", "SVXY"  # Long, Long 2X, Short 1x, Short 0.5x (1x before and including 2018-02-27)
]

# ============================================================
# GICS ETF UNIVERSE
# Primary ETF only per sub-sector (no alternatives)
# Tiers: [A] >$10B  [B] $3-10B  [C] $1-3B  [D] $300M-1B  [E] $100-300M  [F] <$100M
# Spread notes: [E] use limits; [F] trade 10:00-15:30 ET only, verify AUM before live
# ============================================================

# --- GICS Level 1: Sectors ---
sector_ETFs = [
    "XLE",    # Energy                        [A] 1998
    "XLB",    # Materials                     [B] 1998
    "XLI",    # Industrials                   [A] 1998
    "XLY",    # Consumer Discretionary        [A] 1998
    "XLP",    # Consumer Staples              [A] 1998
    "XLV",    # Health Care                   [A] 1998
    "XLF",    # Financials                    [A] 1998
    "XLK",    # Information Technology        [A] 1998
    "XLC",    # Communication Services        [B] 2018
    "XLU",    # Utilities                     [A] 1998
    "XLY",
    "XLRE",
    "VNQ",    # Real Estate                   [A] 2004
]

# --- Energy Sub-Sector (GICS 10) ---
energy_sub_ETFs = [
    "OIH",    # Oil Equipment & Services      [C] 2001  Sub-Industry 10101020
    "XOP",    # Oil & Gas E&P                 [B] 2006  Sub-Industry 10102020
    "CRAK",   # Oil Refiners                  [F] 2015  Sub-Industry 10102030 — watch spreads
    "AMLP",   # MLPs / Pipelines              [B] 2010  Sub-Industry 10102040
]

# --- Materials Sub-Sector (GICS 15) ---
# Note: GDX/GDXJ and SIL/SILJ are NOT alternatives — senior vs junior miners
# are genuinely different size-factor exposures within the same sub-industry
materials_sub_ETFs = [
    "XME",    # Metals & Mining broad         [C] 2006  Industry 151040
    "COPX",   # Copper Miners                 [C] 2010  Sub-Industry 15104025
    "GDX",    # Gold Miners (senior)          [B] 2006  Sub-Industry 15104030
    "GDXJ",   # Gold Miners (junior)          [B] 2009  Sub-Industry 15104030
    "GLD",
    "SIL",    # Silver Miners (senior)        [D] 2010  Sub-Industry 15104045
    "SILJ",   # Silver Miners (junior)        [D] 2012  Sub-Industry 15104045
    "SLV",
    "SLX",    # Steel                         [D] 2006  Sub-Industry 15104050
    "WOOD",   # Timber & Forestry             [D] 2008  Industry 151050 + REIT 60108040
]

# --- Industrials Sub-Sector (GICS 20) ---
industrials_sub_ETFs = [
    "ITA",    # Aerospace & Defense           [B] 2006  Industry 201010
    "PAVE",   # Infrastructure / Const.       [B] 2017  Industry 201030
    "IYT",    # Transportation broad          [C] 2003  Industry Group 2030
    "JETS",   # Airlines                      [C] 2015  Sub-Industry 20302010
    "EVX",    # Environmental Services        [F] 2006  Sub-Industry 20201050 — watch spreads
]

# --- Consumer Discretionary Sub-Sector (GICS 25) ---
consumer_disc_sub_ETFs = [
    "CARZ",   # Automobile Manufacturers      [E] 2011  Sub-Industry 25102010
    "ITB",    # Homebuilders                  [B] 2006  Sub-Industry 25201030
    "XRT",    # Retail broad                  [B] 2006  Industry Groups 2540/2550
    "IBUY",   # Online / E-Commerce Retail    [D] 2016  Sub-Industry 25402010
    "PEJ",    # Hotels, Leisure & Restaurants [D] 2005  Industry 253010
    "BETZ",   # Casinos & Gaming              [D] 2019  Sub-Industry 25301010
]

# --- Consumer Staples Sub-Sector (GICS 30) ---
consumer_staples_sub_ETFs = [
    "PBJ",    # Packaged Foods & Beverages    [D] 2005  Industry 302020
    "MOO",    # Agribusiness                  [C] 2007  Sub-Industry 30202010
]

# --- Health Care Sub-Sector (GICS 35) ---
# WARNING: IBB and XBI are NOT interchangeable — distinct factor exposures:
#   IBB = cap-weighted mega-cap biotech (defensive growth, behaves like XLV)
#   XBI = equal-weighted small/mid biotech (binary FDA-event risk, 50%+ drawdown possible)
healthcare_sub_ETFs = [
    "IHI",    # Medical Devices               [B] 2006  Sub-Industry 35101010
    "IHF",    # HC Providers & Services       [C] 2006  Industry 351020
    "IBB",    # Biotech large-cap             [A] 2001  Sub-Industry 35201010
    "XBI",    # Biotech small/mid-cap         [B] 2006  Sub-Industry 35201010 — different exposure
    "IHE",    # Pharmaceuticals               [D] 2006  Sub-Industry 35202010
]

# --- Financials Sub-Sector (GICS 40) ---
financials_sub_ETFs = [
    "KRE",    # Regional Banks                [B] 2006  Sub-Industry 40101015
    "KBE",    # Diversified Banks broad       [C] 2005  Sub-Industry 40101010
    "REM",    # Mortgage REITs                [C] 2007  Sub-Industry 40204010
    "IAI",    # Capital Markets / Brokers     [C] 2006  Industry 402030
    "KIE",    # Insurance broad               [C] 2005  Industry 403010
    "IPAY",   # Mobile Payments / FinTech     [D] 2015  Sub-Industry 45101020 (bridges IT/Fins)
    "BIZD",   # Business Development Cos.     [D] 2013  BDCs (credit-adjacent Financials)
]

# --- Information Technology Sub-Sector (GICS 45) ---
technology_sub_ETFs = [
    "SOXX",   # Semiconductors                [A] 2001  Sub-Industry 45301020
    "IGV",    # Software (App + Systems)      [B] 2001  Sub-Industries 45102010/20
    "CIBR",   # Cybersecurity                 [C] 2015  Sub-Industry 45102020
    "SKYY",   # Cloud Computing               [C] 2011  Internet Infrastructure
    "ROBO",   # Robotics & AI                 [C] 2013  GICS 20106020 + 45102010
]

# --- Communication Services Sub-Sector (GICS 50) ---
comm_services_sub_ETFs = [
    "IYZ",    # Telecom (legacy: T/VZ/TMUS)   [C] 2000  Industry 501010
    "ESPO",   # Video Gaming & Esports         [D] 2018  Sub-Industry 50202020
]

# --- Utilities Sub-Sector (GICS 55) ---
# Note: TAN and FAN are both Sub-Industry 55105020 (Renewable Electricity)
# but target distinct generation technologies — not alternatives
utilities_sub_ETFs = [
    "PHO",    # Water Utilities               [C] 2005  Sub-Industry 55104010
    "ICLN",   # Clean / Renewable Energy      [B] 2008  Sub-Industry 55105020 (broad)
    "TAN",    # Solar                         [C] 2008  Sub-Industry 55105020 (solar)
    "FAN",    # Wind                          [D] 2008  Sub-Industry 55105020 (wind)
]

# --- Real Estate Sub-Sector (GICS 60) ---
# WOOD (Timber REITs, Sub-Industry 60108040) already captured in materials_sub_ETFs
real_estate_sub_ETFs = [
    "INDS",   # Industrial REITs              [D] 2018  Sub-Industry 60102510
    "REZ",    # Residential REITs             [D] 2007  Sub-Industry 60106010/20
    "SRVR",   # Data Centre & Tower REITs     [D] 2018  Sub-Industry 60108050
]

# ============================================================
# Combined Universes
# ============================================================

gics_sub_ETFs = list(set(
    energy_sub_ETFs +
    materials_sub_ETFs +
    industrials_sub_ETFs +
    consumer_disc_sub_ETFs +
    consumer_staples_sub_ETFs +
    healthcare_sub_ETFs +
    financials_sub_ETFs +
    technology_sub_ETFs +
    comm_services_sub_ETFs +
    utilities_sub_ETFs +
    real_estate_sub_ETFs
))  # 52 unique sub-sector ETFs

gics_all_ETFs = list(set(sector_ETFs + gics_sub_ETFs))  # 63 total

In [3]:
all_ETFs = list(set(
    size_ETFs + 
    factor_ETFs + 
    bond_ETFs + 
    region_ETFs + subregion_ETFs + country_ETFs + 
    leveraged_ETFs + 
    inverse_ETFs +
    volatility_ETFs + 
    sector_ETFs +
    gics_sub_ETFs
))

In [4]:
print(f"Total Unique Highly Liquid ETFs in Universe: {len(all_ETFs)}")

Total Unique Highly Liquid ETFs in Universe: 171


In [5]:
# All available ETF tickers

ticker_list_ETF = client.list_tickers(type="ETF", date=END_DATE.isoformat(), active=True, market='stocks', limit=1000)
ticker_list_ETN = client.list_tickers(type="ETN", date=END_DATE.isoformat(), active=True, market='stocks', limit=1000)
ticker_list_ETV = client.list_tickers(type="ETV", date=END_DATE.isoformat(), active=True, market='stocks', limit=1000)

ticker_list_ETF = pd.DataFrame(ticker_list_ETF)
ticker_list_ETN = pd.DataFrame(ticker_list_ETN)
ticker_list_ETV = pd.DataFrame(ticker_list_ETV)

tickers_ETP = pd.concat([ticker_list_ETF, ticker_list_ETN, ticker_list_ETV]).reset_index()

In [6]:
print(len(tickers_ETP))
tickers_ETP.head(3)

5225


,index,active,cik,composite_figi,currency_name,currency_symbol,base_currency_symbol,base_currency_name,delisted_utc,last_updated_utc,locale,market,name,primary_exchange,share_class_figi,ticker,type,source_feed
0,0,True,0001776878,BBG01B0JRCS6,usd,None,None,None,None,2026-06-04T06:08:59.302777369Z,us,stocks,Alternative Access First Priority CLO Bond ETF,ARCX,BBG01B0JRCT5,AAA,ETF,None
1,1,True,NaN,BBG01W275XX6,usd,None,None,None,None,2026-06-04T06:08:59.302777479Z,us,stocks,Amplius Aggressive Asset Allocation ETF,BATS,BBG01W275ZB5,AAAA,ETF,None
2,2,True,0001551950,BBG01YZNRVY0,usd,None,None,None,None,2026-06-04T06:08:59.30277756Z,us,stocks,Columbia AAA CLO ETF,ARCX,BBG01YZNRWV1,AAAC,ETF,None


In [7]:
# Duplicates?

tickers_ETP_unique = tickers_ETP[~tickers_ETP['ticker'].duplicated()]
print(len(tickers_ETP_unique))


5225


# Only get data for the major 150-ish ETFs that we are interested in. 



In [8]:
tickers_ETP = tickers_ETP[tickers_ETP["ticker"].isin(all_ETFs)]

print(len(tickers_ETP))
tickers_ETP.head(3)

171


,index,active,cik,composite_figi,currency_name,currency_symbol,base_currency_symbol,base_currency_name,delisted_utc,last_updated_utc,locale,market,name,primary_exchange,share_class_figi,ticker,type,source_feed
21,21,True,0001100663,BBG000G6GXC5,usd,None,None,None,None,2026-06-04T06:08:59.302782529Z,us,stocks,iShares MSCI All Country Asia ex Japan ETF,XNAS,BBG001T2V2D8,AAXJ,ETF,None
76,76,True,0001100663,BBG000Q123R0,usd,None,None,None,None,2026-06-04T06:08:59.302838606Z,us,stocks,iShares Core U.S. Aggregate Bond ETF,ARCX,BBG001SM1QT8,AGG,ETF,None
133,133,True,0001414040,BBG0015CS200,usd,None,None,None,None,2026-06-04T06:08:59.302870306Z,us,stocks,Alerian MLP ETF,ARCX,BBG001TBMNT5,AMLP,ETF,None


In [9]:
tickers_ETP = tickers_ETP[['ticker', 'name', 'active', 'cik', 'composite_figi', 'type']]
tickers_ETP = tickers_ETP.reindex(columns = ["ID", "ticker", "name", "active", "start_date", "end_date", "type", "cik", "composite_figi"]) 

tickers_ETP['start_date'] = pd.to_datetime(tickers_ETP['start_date']).dt.date
tickers_ETP["end_date"] = pd.to_datetime(tickers_ETP["end_date"]).dt.date

tickers_ETP.reset_index(drop=True, inplace=True)

print(len(tickers_ETP))
tickers_ETP.head(3)

171


,ID,ticker,name,active,start_date,end_date,type,cik,composite_figi
0,NaN,AAXJ,iShares MSCI All Country Asia ex Japan ETF,True,NaT,NaT,ETF,0001100663,BBG000G6GXC5
1,NaN,AGG,iShares Core U.S. Aggregate Bond ETF,True,NaT,NaT,ETF,0001100663,BBG000Q123R0
2,NaN,AMLP,Alerian MLP ETF,True,NaT,NaT,ETF,0001414040,BBG0015CS200


In [10]:
missing_etf = set(all_ETFs) - set(tickers_ETP["ticker"])
print(f"Missing ETF: {missing_etf}")

Missing ETF: set()


Here we fill in the blanks using the <code>get_tickers_events</code> function. We need the start date from the ticker, which may not always be equal to START_DATE. I assume no delistings.

In [11]:
print(tickers_ETP['start_date'].dtype)

datetime64[s]


In [12]:
""" for index, row in tickers_ETP.copy().iterrows():
    if row["type"] == "ETF" or row["type"] == "ETN" or row["type"] == "ETV":
        xyz = tickers_ETP.columns.get_loc("start_date")
        print(xyz)
        print(type(xyz)) """

' for index, row in tickers_ETP.copy().iterrows():\n    if row["type"] == "ETF" or row["type"] == "ETN" or row["type"] == "ETV":\n        xyz = tickers_ETP.columns.get_loc("start_date")\n        print(xyz)\n        print(type(xyz)) '

In [13]:
""" for index, row in tickers_ETP.copy().iterrows():
    if row["type"] == "ETF" or row["type"] == "ETN" or row["type"] == "ETV":
        test = max(date.fromisoformat(client.get_ticker_events(row["ticker"]).events[-1]['date']), START_DATE)
        print(test)
        print(type(test))
        tickers_ETP.loc[index, 'start_date'] = pd.Timestamp(test)
 """

' for index, row in tickers_ETP.copy().iterrows():\n    if row["type"] == "ETF" or row["type"] == "ETN" or row["type"] == "ETV":\n        test = max(date.fromisoformat(client.get_ticker_events(row["ticker"]).events[-1][\'date\']), START_DATE)\n        print(test)\n        print(type(test))\n        tickers_ETP.loc[index, \'start_date\'] = pd.Timestamp(test)\n '

In [14]:
""" idx = tickers_ETP.columns.get_loc("start_date")
print(tickers_ETP.columns[idx])  # Should print 'start_date' """

' idx = tickers_ETP.columns.get_loc("start_date")\nprint(tickers_ETP.columns[idx])  # Should print \'start_date\' '

In [15]:
""" for index, row in tickers_ETP.copy().iterrows():
    if row["type"] in ["ETF", "ETN", "ETV"]:
        result = tickers_ETP.loc[index, "start_date"]
        print(result) """

' for index, row in tickers_ETP.copy().iterrows():\n    if row["type"] in ["ETF", "ETN", "ETV"]:\n        result = tickers_ETP.loc[index, "start_date"]\n        print(result) '

Pandas cannot automatically convert a date object into a datetime64[s] scalar, hence the TypeError.

In [16]:
# Ensure start_date column can hold Python date objects
tickers_ETP['start_date'] = tickers_ETP['start_date'].astype(object)

for index, row in tickers_ETP.copy().iterrows():
    if row["type"] in ("ETF", "ETN", "ETV"):
        try:
            last_event_date = date.fromisoformat(client.get_ticker_events(row["ticker"]).events[-1]['date'])
            # print(last_event_date)
            new_date = max(last_event_date, START_DATE)
            # print(new_date)
            tickers_ETP.at[index, 'start_date'] = new_date
            # ✅ Print the updated value from the original DataFrame
            # print(tickers_ETP.at[index, 'start_date'])   # This will show the actual date
            # print("Done for this row")
        except Exception as e:
            tickers_ETP.at[index, 'start_date'] = START_DATE
            print(row["ticker"])
            print(row["start_date"])
            print(repr(e))

In [17]:
tickers_ETP

,ID,ticker,name,active,start_date,end_date,type,cik,composite_figi
0,NaN,AAXJ,iShares MSCI All Country Asia ex Japan ETF,True,2016-06-06,NaT,ETF,0001100663,BBG000G6GXC5
1,NaN,AGG,iShares Core U.S. Aggregate Bond ETF,True,2016-06-06,NaT,ETF,0001100663,BBG000Q123R0
2,NaN,AMLP,Alerian MLP ETF,True,2020-05-18,NaT,ETF,0001414040,BBG0015CS200
3,NaN,BETZ,Roundhill Sports Betting & iGaming ETF,True,2020-06-04,NaT,ETF,0001776878,BBG00V7866V2
4,NaN,BIL,State Street SPDR Bloomberg 1-3 Month T-Bill ETF,True,2017-11-30,NaT,ETF,0001081019,BBG000RFQSH8
...,...,...,...,...,...,...,...,...,...
166,NaN,XOP,State Street SPDR S&P Oil & Gas Exploration & ...,True,2020-03-30,NaT,ETF,0001081019,BBG000BGB482
167,NaN,XRT,State Street SPDR S&P Retail ETF,True,2016-06-06,NaT,ETF,0001064642,BBG000D80VV4
168,NaN,GLD,"SPDR Gold Trust, SPDR Gold Shares",True,2016-06-06,NaT,ETV,0001222333,BBG000CRF6Q8
169,NaN,SLV,iShares Silver Trust,True,2016-06-06,NaT,ETV,0001330568,BBG000NDCRW7


In [18]:
""" for index, row in tickers_ETP.copy().iterrows():
    if row["type"] == "ETF" or row["type"] == "ETN" or row["type"] == "ETV":
        try:
            tickers_ETP.iloc[index, tickers_ETP.columns.get_loc("start_date")] = max(date.fromisoformat(client.get_ticker_events(row["ticker"]).events[-1]['date']), START_DATE)
            print(row["start_date"])
        except Exception as e:
            # tickers_ETP.iloc[index, tickers_ETP.columns.get_loc("start_date")] = START_DATE
            print(row["ticker"])
            print(row["start_date"])
            print(repr(e))
 """

# Set remaining columns
# tickers_ETP["end_date"].fillna(END_DATE, inplace=True)
# tickers_ETP["ID"] = tickers_ETP["ticker"] + '-' + tickers_ETP["start_date"].astype(str)

' for index, row in tickers_ETP.copy().iterrows():\n    if row["type"] == "ETF" or row["type"] == "ETN" or row["type"] == "ETV":\n        try:\n            tickers_ETP.iloc[index, tickers_ETP.columns.get_loc("start_date")] = max(date.fromisoformat(client.get_ticker_events(row["ticker"]).events[-1][\'date\']), START_DATE)\n            print(row["start_date"])\n        except Exception as e:\n            # tickers_ETP.iloc[index, tickers_ETP.columns.get_loc("start_date")] = START_DATE\n            print(row["ticker"])\n            print(row["start_date"])\n            print(repr(e))\n '

In [19]:
""" for index, row in tickers_ETP.copy().iterrows():
    if row["type"] in ["ETF", "ETN", "ETV"]:
        try:
            relevant_date = max(date.fromisoformat(client.get_ticker_events(row["ticker"]).events[-1]['date']), START_DATE)
            # tickers_ETP.loc[index, "start_date"] = pd.Timestamp(relevant_date)
            tickers_ETP.loc[index, "start_date"] = pd.Timestamp(relevant_date)
            # tickers_ETP.loc[index, "start_date"] = pd.to_datetime(relevant_date).dt.date
            # tickers_ETP['start_date'] = pd.to_datetime(tickers_ETP['start_date']).dt.date
            print(row["start_date"])
        except Exception as e:
            tickers_ETP.loc[index, "start_date"] = pd.Timestamp(START_DATE)
            print(row["ticker"])
            print(repr(e)) """

' for index, row in tickers_ETP.copy().iterrows():\n    if row["type"] in ["ETF", "ETN", "ETV"]:\n        try:\n            relevant_date = max(date.fromisoformat(client.get_ticker_events(row["ticker"]).events[-1][\'date\']), START_DATE)\n            # tickers_ETP.loc[index, "start_date"] = pd.Timestamp(relevant_date)\n            tickers_ETP.loc[index, "start_date"] = pd.Timestamp(relevant_date)\n            # tickers_ETP.loc[index, "start_date"] = pd.to_datetime(relevant_date).dt.date\n            # tickers_ETP[\'start_date\'] = pd.to_datetime(tickers_ETP[\'start_date\']).dt.date\n            print(row["start_date"])\n        except Exception as e:\n            tickers_ETP.loc[index, "start_date"] = pd.Timestamp(START_DATE)\n            print(row["ticker"])\n            print(repr(e)) '

In [20]:
# Set remaining columns
tickers_ETP['end_date'] = tickers_ETP['end_date'].astype(object)

tickers_ETP["end_date"] = tickers_ETP["end_date"].fillna(END_DATE)

tickers_ETP["ID"] = tickers_ETP["ticker"] + '-' + tickers_ETP["start_date"].astype(str)

tickers_ETP.head(3)

,ID,ticker,name,active,start_date,end_date,type,cik,composite_figi
0,AAXJ-2016-06-06,AAXJ,iShares MSCI All Country Asia ex Japan ETF,True,2016-06-06,2026-06-03,ETF,0001100663,BBG000G6GXC5
1,AGG-2016-06-06,AGG,iShares Core U.S. Aggregate Bond ETF,True,2016-06-06,2026-06-03,ETF,0001100663,BBG000Q123R0
2,AMLP-2020-05-18,AMLP,Alerian MLP ETF,True,2020-05-18,2026-06-03,ETF,0001414040,BBG0015CS200


In [21]:
""" for index, row in tickers_ETP.copy().iterrows():
    if row["type"] in ["ETF", "ETN", "ETV"]:
        try:
            print(index)
            print(row)
            relevant_date = max(date.fromisoformat(client.get_ticker_events(row["ticker"]).events[-1]['date']), START_DATE)
            print(pd.Timestamp(relevant_date))
            tickers_ETP.loc[index, tickers_ETP.columns.get_loc("start_date")] = pd.Timestamp(relevant_date)
        except Exception as e:
            tickers_ETP.loc[index, tickers_ETP.columns.get_loc("start_date")] = pd.Timestamp(START_DATE)
            print(row["ticker"])
            print(repr(e))

# Set remaining columns
tickers_ETP["end_date"].fillna(END_DATE, inplace=True)
tickers_ETP["ID"] = tickers_ETP["ticker"] + '-' + tickers_ETP["start_date"].astype(str) """

' for index, row in tickers_ETP.copy().iterrows():\n    if row["type"] in ["ETF", "ETN", "ETV"]:\n        try:\n            print(index)\n            print(row)\n            relevant_date = max(date.fromisoformat(client.get_ticker_events(row["ticker"]).events[-1][\'date\']), START_DATE)\n            print(pd.Timestamp(relevant_date))\n            tickers_ETP.loc[index, tickers_ETP.columns.get_loc("start_date")] = pd.Timestamp(relevant_date)\n        except Exception as e:\n            tickers_ETP.loc[index, tickers_ETP.columns.get_loc("start_date")] = pd.Timestamp(START_DATE)\n            print(row["ticker"])\n            print(repr(e))\n\n# Set remaining columns\ntickers_ETP["end_date"].fillna(END_DATE, inplace=True)\ntickers_ETP["ID"] = tickers_ETP["ticker"] + \'-\' + tickers_ETP["start_date"].astype(str) '

In [22]:
""" for index, row in tickers_ETP.copy().iterrows():
    if row["type"] == "ETF" or row["type"] == "ETN" or row["type"] == "ETV":
        try:
            # tickers_ETP.iloc[index, tickers_ETP.columns.get_loc("start_date")] = max(date.fromisoformat(client.get_ticker_events(row["ticker"]).events[-1]['date']), START_DATE)
            tickers_ETP.loc[index, tickers_ETP.columns.get_loc("start_date")] = max(date.fromisoformat(client.get_ticker_events(row["ticker"]).events[-1]['date']), START_DATE)
        except Exception as e:
            tickers_ETP.loc[index, tickers_ETP.columns.get_loc("start_date")] = START_DATE
            print(row["ticker"])
            print(repr(e))

# Set remaining columns
tickers_ETP["end_date"].fillna(END_DATE, inplace=True)
tickers_ETP["ID"] = tickers_ETP["ticker"] + '-' + tickers_ETP["start_date"].astype(str) """

' for index, row in tickers_ETP.copy().iterrows():\n    if row["type"] == "ETF" or row["type"] == "ETN" or row["type"] == "ETV":\n        try:\n            # tickers_ETP.iloc[index, tickers_ETP.columns.get_loc("start_date")] = max(date.fromisoformat(client.get_ticker_events(row["ticker"]).events[-1][\'date\']), START_DATE)\n            tickers_ETP.loc[index, tickers_ETP.columns.get_loc("start_date")] = max(date.fromisoformat(client.get_ticker_events(row["ticker"]).events[-1][\'date\']), START_DATE)\n        except Exception as e:\n            tickers_ETP.loc[index, tickers_ETP.columns.get_loc("start_date")] = START_DATE\n            print(row["ticker"])\n            print(repr(e))\n\n# Set remaining columns\ntickers_ETP["end_date"].fillna(END_DATE, inplace=True)\ntickers_ETP["ID"] = tickers_ETP["ticker"] + \'-\' + tickers_ETP["start_date"].astype(str) '

We need to manually change VXX and UVIX start date. The <code>get_ticker_events</code> is incorrect.

In [23]:
tickers_ETP[tickers_ETP['ticker'] == 'VXX']

,ID,ticker,name,active,start_date,end_date,type,cik,composite_figi
150,VXX-2024-07-24,VXX,iPath Series B S&P 500 VIX Short-Term Futures ETN,True,2024-07-24,2026-06-03,ETF,0000312070,BBG00JQ5JWB5


In [24]:
tickers_ETP[tickers_ETP['ticker'] == 'UVIX']

,ID,ticker,name,active,start_date,end_date,type,cik,composite_figi
132,UVIX-2025-01-15,UVIX,2x Long VIX Futures ETF,True,2025-01-15,2026-06-03,ETF,0001793497,BBG014HVGDK5


In [25]:
# tickers_ETP.loc[tickers_ETP['ticker'] == 'VXX', "start_date"] = max(date(2009, 1, 29), START_DATE)
# tickers_ETP.loc[tickers_ETP['ticker'] == 'UVIX', "start_date"] = max(date(2022, 3, 30), START_DATE)

tickers_ETP.loc[tickers_ETP['ticker'] == 'VXX', "start_date"] = max(date(2009, 1, 29), START_DATE)
tickers_ETP.loc[tickers_ETP['ticker'] == 'UVIX', "start_date"] = max(date(2022, 3, 30), START_DATE)

In [26]:
tickers_ETP[tickers_ETP['ticker'] == 'VXX']

,ID,ticker,name,active,start_date,end_date,type,cik,composite_figi
150,VXX-2024-07-24,VXX,iPath Series B S&P 500 VIX Short-Term Futures ETN,True,2016-06-06,2026-06-03,ETF,0000312070,BBG00JQ5JWB5


In [27]:
tickers_ETP[tickers_ETP['ticker'] == 'UVIX']

,ID,ticker,name,active,start_date,end_date,type,cik,composite_figi
132,UVIX-2025-01-15,UVIX,2x Long VIX Futures ETF,True,2022-03-30,2026-06-03,ETF,0001793497,BBG014HVGDK5


I will call everything an 'ETF' for simplicity.

In [28]:
tickers_ETP['type'] = 'ETF'

In [29]:
# Merge tickers_v1 and ETFs to get tickers_v1
tickers_v1 = get_tickers(v=1)
tickers_v2 = pd.concat([tickers_v1, tickers_ETP])
tickers_v2.reset_index(drop=True, inplace=True)
tickers_v2.tail(3)

,ID,ticker,name,active,start_date,end_date,type,cik,composite_figi
11283,GLD-2016-06-06,GLD,"SPDR Gold Trust, SPDR Gold Shares",True,2016-06-06,2026-06-03,ETF,0001222333,BBG000CRF6Q8
11284,SLV-2016-06-06,SLV,iShares Silver Trust,True,2016-06-06,2026-06-03,ETF,0001330568,BBG000NDCRW7
11285,UCO-2020-04-21,UCO,ProShares Ultra Bloomberg Crude Oil,True,2020-04-21,2026-06-03,ETF,0001415311,BBG000CSVPZ6


In [30]:
tickers_v2.head(3)

,ID,ticker,name,active,start_date,end_date,type,cik,composite_figi
0,A-2016-06-06,A,Agilent Technologies Inc.,True,2016-06-06,2026-06-03,CS,1090872.0,BBG000C2V3D6
1,AA-2016-06-06,AA,Alcoa Corporation,True,2016-06-06,2026-06-03,CS,1675149.0,BBG00B3T3HD3
2,AAAP-2016-06-06,AAAP,Advanced Accelerator Applications S.A. America...,False,2016-06-06,2018-02-09,ADRC,1611787.0,BBG00K6FMBQ8


In [35]:
tickers_v2.to_csv("../data/tickers_v2.csv") 

In [ ]:
assert False

# 4.2 Updates
We simply copy/paste the ETPs from the old to the new ticker_v3.

In [ ]:
old_tickers_v2 = get_tickers(2)
ETPs = tickers_v2[tickers_v2['type'] == 'ETF']
ETPs.loc[:, 'end_date'] = END_DATE # I assume that no ETFs got delisted/renamed.
pd.concat([get_tickers(2), ETPs]).reset_index(drop=True).to_csv("../data/tickers_v2.csv")

# ETF Dividends matter (VERIFY that they get download during "05_download_adjustments.ipynb" and applied in 07_process.ipynb)
When querying historical data for these via Yahoo Finance (yfinance), Polygon, or Alpaca, make sure you pull Adjusted Close (Adj Close). ETFs like HYG (Junk Bonds) and TLT (Treasuries) pay hefty monthly dividends. If your code uses raw Close prices instead of Adjusted Close prices, your moving averages will trigger fake breakdown signals every time the ETF goes ex-dividend.

# GICS ETF Coverage Map — US Equities

**Note on sub-industry count:** Current GICS (2023 revision) has 163 sub-industries, up from the pre-2023 count of 145. Coverage density drops sharply as you go deeper. The sweet spot for ETF-based systematic strategies is the Industry Group and Industry levels.

**Liquidity tiers (approximate AUM):**
- `[A]` = >$10B
- `[B]` = $3–10B
- `[C]` = $1–3B
- `[D]` = $300M–1B
- `[E]` = $100–300M
- `[F]` = <$100M — watch spreads carefully

---

## Master Sector Table

| GICS | Sector | Primary | Tier | Inception | Alternatives |
|------|--------|---------|------|-----------|--------------|
| 10 | Energy | XLE | A | 1998 | VDE (2004, B), IYE (2000, C) |
| 15 | Materials | XLB | B | 1998 | VAW (2004, B), IYM (2000, C) |
| 20 | Industrials | XLI | A | 1998 | VIS (2004, B), IYJ (2000, C) |
| 25 | Consumer Discretionary | XLY | A | 1998 | VCR (2004, C), IYC (2000, C) |
| 30 | Consumer Staples | XLP | A | 1998 | VDC (2004, B), IYK (2000, C) |
| 35 | Health Care | XLV | A | 1998 | VHT (2004, B), IYH (2000, B) |
| 40 | Financials | XLF | A | 1998 | VFH (2004, B), IYF (2000, C) |
| 45 | Information Technology | XLK | A | 1998 | VGT (2004, A), IYW (2000, B), FTEC (2013, B) |
| 50 | Communication Services | XLC | B | 2018 | VOX (2004†, C), FCOM (2013, C) |
| 55 | Utilities | XLU | A | 1998 | VPU (2004, B), IDU (2000, C) |
| 60 | Real Estate | VNQ | A | 2004 | XLRE (2015, B), IYR (2000, B), RWR (2001, C) |

† VOX tracked Telecom-only 2004–2018, restructured to Communication Services when GICS revised in 2018.

---

## Sector 10 — Energy

Sector ETFs: **XLE** [A, 1998], VDE [B, 2004], IYE [C, 2000]

### Industry Group 1010: Energy

**Industry 101010 — Energy Equipment & Services**

| Sub-Industry | Code | ETF | Tier | Year | Notes |
|---|---|---|---|---|---|
| Oil & Gas Equipment & Services | 10101020 | **OIH** VanEck Oil Services | C | 2001 | Oldest & most liquid oil services ETF |
| Oil & Gas Equipment & Services | 10101020 | IEZ iShares US Oil Equipment | D | 2006 | Alt |
| Oil & Gas Drilling | 10101010 | — | — | — | No dedicated liquid ETF |

**Industry 101020 — Oil, Gas & Consumable Fuels**

| Sub-Industry | Code | ETF | Tier | Year | Notes |
|---|---|---|---|---|---|
| E&P | 10102020 | **XOP** SPDR S&P Oil & Gas E&P | B | 2006 | Equal-weighted, high beta, most liquid E&P ETF |
| Refining | 10102030 | **CRAK** VanEck Oil Refiners | F | 2015 | Only dedicated US refiner ETF; small AUM, verify spreads |
| Pipelines/MLPs | 10102040 | **AMLP** Alerian MLP | B | 2010 | Largest MLP ETF; C-corp tax structure impacts returns |
| Pipelines/MLPs | 10102040 | MLPA Global X MLP | D | 2012 | ETF (not ETN) pass-through structure |
| Pipelines/MLPs | 10102040 | MLPX Global X MLP & Energy Infrastructure | D | 2013 | Broader: MLPs + C-corp pipelines |
| Integrated Oil | 10102010 | — | — | — | No dedicated ETF; XLE/VDE proxy |
| Coal | 10102050 | — | — | — | KOL (VanEck Coal) liquidated 2020; no liquid replacement |

---

## Sector 15 — Materials

Sector ETFs: **XLB** [B, 1998], VAW [B, 2004], IYM [C, 2000]

### Industry Group 1510: Materials

**Industry 151010 — Chemicals**

No dedicated liquid US ETF for any chemicals sub-industry. MOO (see below) partially covers Fertilizers & Agricultural Chemicals (15101030).

**Industry 151040 — Metals & Mining**

| Sub-Industry | Code | ETF | Tier | Year | Notes |
|---|---|---|---|---|---|
| Diversified Metals & Mining | 15104020 | **XME** SPDR S&P Metals & Mining | C | 2006 | Most liquid US metals & mining ETF |
| Diversified Metals & Mining | 15104020 | PICK iShares MSCI Global Metals & Mining | D | 2012 | Global exposure |
| Copper | 15104025 | **COPX** Global X Copper Miners | C | 2010 | Only dedicated copper miners ETF |
| Gold | 15104030 | **GDX** VanEck Gold Miners | B | 2006 | Most liquid gold miners ETF globally |
| Gold (junior) | 15104030 | GDXJ VanEck Junior Gold Miners | B | 2009 | Smaller miners, higher volatility |
| Silver | 15104045 | **SIL** Global X Silver Miners | D | 2010 | |
| Silver (junior) | 15104045 | SILJ ETFMG Prime Junior Silver Miners | D | 2012 | |
| Steel | 15104050 | **SLX** VanEck Steel | D | 2006 | |

**Industry 151050 — Paper & Forest Products**

| Sub-Industry | Code | ETF | Tier | Year | Notes |
|---|---|---|---|---|---|
| Forest Products | 15105010 | **WOOD** iShares Global Timber & Forestry | D | 2008 | Also covers Timber REITs (GICS 60108040) |
| Forest Products | 15105010 | CUT Invesco MSCI Global Timber | E | 2008 | Alt |
| Paper Products | 15105020 | — | — | — | No dedicated ETF |

**Industries 151020, 151030 — Construction Materials; Containers & Packaging:** No dedicated liquid ETF for any sub-industry. XLB is the sector proxy.

---

## Sector 20 — Industrials

Sector ETFs: **XLI** [A, 1998], VIS [B, 2004], IYJ [C, 2000]

### Industry Group 2010: Capital Goods

**Industry 201010 — Aerospace & Defense (Sub-Industry 20101010)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **ITA** iShares US Aerospace & Defense | B | 2006 | Most liquid A&D ETF, market-cap weighted |
| PPA Invesco Aerospace & Defense | D | 2005 | Oldest A&D ETF |
| XAR SPDR S&P Aerospace & Defense | D | 2011 | Equal-weighted |

**Industry 201030 — Construction & Engineering (Sub-Industry 20103010)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **PAVE** Global X US Infrastructure Development | B | 2017 | Best proxy; spans construction, building products, machinery |

**Industries 201020, 201040, 201060, 201070 — Building Products; Electrical Equipment; Machinery; Trading Companies:** No dedicated liquid ETF for any sub-industry. XLI/PAVE as proxies.

### Industry Group 2020: Commercial & Professional Services

**Industry 202010 — Commercial Services & Supplies**

| Sub-Industry | Code | ETF | Tier | Year | Notes |
|---|---|---|---|---|---|
| Environmental & Facilities Services | 20201050 | **EVX** VanEck Environmental Services | F | 2006 | Small AUM — verify spreads before trading |
| All other sub-industries | — | — | — | — | No dedicated ETF |

**Industry 202020 — Professional Services:** No dedicated ETF for any sub-industry.

### Industry Group 2030: Transportation

Broad transport: **IYT** iShares Transportation Average [C, 2003] (oldest), XTN SPDR S&P Transportation [D, 2011] (equal-weighted)

**Industry 203020 — Passenger Airlines (Sub-Industry 20302010)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **JETS** US Global Jets | C | 2015 | Only dedicated airline ETF |

**Industries 203010, 203030–203050 — Air Freight; Marine; Ground Transportation; Transport Infrastructure:** No dedicated liquid US ETF. IYT is the proxy for air freight (heavy UPS/FedEx/railroads weighting).

---

## Sector 25 — Consumer Discretionary

Sector ETFs: **XLY** [A, 1998], VCR [C, 2004], IYC [C, 2000]

### Industry Group 2510: Automobiles & Components

**Industry 251020 — Automobiles (Sub-Industry 25102010)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **CARZ** First Trust NASDAQ Global Auto | E | 2011 | Pure auto manufacturers |
| DRIV Global X Autonomous & Electric Vehicles | D | 2018 | Broader: auto + semiconductors + software |

**Industry 251010 — Automobile Components:** No dedicated ETF.

### Industry Group 2520: Consumer Durables & Apparel

**Industry 252010 — Household Durables, Sub-Industry 25201030 Homebuilding**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **ITB** iShares US Home Construction | B | 2006 | Most liquid homebuilder ETF, market-cap weighted |
| XHB SPDR S&P Homebuilders | C | 2006 | Equal-weighted; broader supply chain inc. appliances, flooring |

**All other Consumer Durables & Apparel sub-industries (Electronics, Apparel, Footwear, Luxury, Textiles):** No dedicated liquid US ETF.

### Industry Group 2530: Consumer Services

**Industry 253010 — Hotels, Restaurants & Leisure**

| Sub-Industry | Code | ETF | Tier | Year | Notes |
|---|---|---|---|---|---|
| Casinos & Gaming | 25301010 | **BETZ** Roundhill Sports Betting & iGaming | D | 2019 | Most liquid gaming ETF, US-skewed |
| Casinos & Gaming | 25301010 | BJK VanEck Gaming | E | 2008 | Oldest gaming ETF, global-focused |
| Hotels, Resorts & Cruise | 25301020 | **PEJ** Invesco Dynamic Leisure & Entertainment | D | 2005 | Oldest leisure ETF: hotels + restaurants + leisure |
| Hotels, Resorts & Cruise | 25301020 | AWAY Defiance Hotel, Airline & Cruise | E | 2021 | Smaller, newer |
| Restaurants | 25301040 | — | — | — | No pure restaurant ETF; PEJ is closest proxy |
| Leisure Facilities | 25301030 | — | — | — | No dedicated ETF |

**Industry 253020 — Diversified Consumer Services:** No dedicated ETF.

### Industry Group 2540/2550: Retailing

**Broad Retail (all retail sub-industries)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **XRT** SPDR S&P Retail | B | 2006 | Most liquid retail ETF, equal-weighted ~80 names |
| RTH VanEck Retail | C | 2001 | Oldest retail ETF, large-cap weighted |

**E-Commerce / Internet Retail (Sub-Industry 25402010)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **IBUY** Amplify Online Retail | D | 2016 | Most liquid online retail ETF |
| ONLN ProShares Online Retail | E | 2018 | |

---

## Sector 30 — Consumer Staples

Sector ETFs: **XLP** [A, 1998], VDC [B, 2004], IYK [C, 2000]

### Industry Group 3010: Food & Staples Retailing

Sub-Industries 30101010–30101040 (Drug Retail, Food Distributors, Food Retail, Consumer Staples Merchandise Retail): No dedicated ETF for any. XLP is the proxy.

### Industry Group 3020: Food, Beverage & Tobacco

**Industry 302020 — Food Products**

| Sub-Industry | Code | ETF | Tier | Year | Notes |
|---|---|---|---|---|---|
| Packaged Foods & Meats | 30202030 | **PBJ** Invesco Dynamic Food & Beverage | D | 2005 | Only dedicated food & beverage ETF |
| Agricultural Products | 30202010 | **MOO** VanEck Agribusiness | C | 2007 | Ag chemicals, farm equipment, food processing |

**Industry 302010 — Beverages (Brewers, Distillers, Soft Drinks):** No dedicated liquid US ETF for any sub-industry.

**Industry 302030 — Tobacco:** No dedicated liquid US ETF.

### Industry Group 3030: Household & Personal Products

Sub-Industries 30301010 and 30302010: No dedicated ETF. XLP proxy.

---

## Sector 35 — Health Care

Sector ETFs: **XLV** [A, 1998], VHT [B, 2004], IYH [B, 2000]

### Industry Group 3510: Health Care Equipment & Services

**Industry 351010 — Health Care Equipment & Supplies**

| Sub-Industry | Code | ETF | Tier | Year | Notes |
|---|---|---|---|---|---|
| Health Care Equipment | 35101010 | **IHI** iShares US Medical Devices | B | 2006 | Most liquid medical devices ETF |
| Health Care Supplies | 35101020 | — | — | — | No dedicated ETF |

**Industry 351020 — Health Care Providers & Services**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **IHF** iShares US Healthcare Providers | C | 2006 | Covers managed care, HC facilities, distributors (35102010–35102040) |

**Industry 351030 — Health Care Technology (Sub-Industry 35103010):** No dedicated liquid US ETF.

### Industry Group 3520: Pharmaceuticals, Biotechnology & Life Sciences

**Industry 352010 — Biotechnology (Sub-Industry 35201010)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **XBI** SPDR S&P Biotech | B | 2006 | Extremely liquid; equal-weighted ~150 names; massive options market; small/mid-cap dominated |
| **IBB** iShares Nasdaq Biotechnology | A | 2001 | Oldest biotech ETF; market-cap weighted — dominated by AMGN/GILD/VRTX; behaves very differently from XBI |

Do not treat XBI and IBB as substitutes. They are different factor bets: XBI is binary-event small-cap exposure, IBB is mega-cap defensive biotech.

**Industry 352020 — Pharmaceuticals (Sub-Industry 35202010)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **IHE** iShares US Pharmaceuticals | D | 2006 | |
| XPH SPDR S&P Pharmaceuticals | D | 2006 | Equal-weighted |
| PPH VanEck Pharmaceutical | D | 2011 | Large-cap global pharma |

**Industry 352030 — Life Sciences Tools & Services (Sub-Industry 35203010):** No dedicated liquid US ETF.

---

## Sector 40 — Financials

Sector ETFs: **XLF** [A, 1998], VFH [B, 2004], IYF [C, 2000]

### Industry Group 4010: Banks

**Industry 401010 — Banks**

| Sub-Industry | Code | ETF | Tier | Year | Notes |
|---|---|---|---|---|---|
| Diversified Banks | 40101010 | **KBE** SPDR S&P Bank | C | 2005 | Broad banks, equal-weighted |
| Diversified Banks | 40101010 | KBWB Invesco KBW Bank | C | 2011 | Large-cap banks, market-cap weighted |
| Regional Banks | 40101015 | **KRE** SPDR S&P Regional Banking | B | 2006 | Most liquid regional bank ETF; very actively traded around FOMC |
| Regional Banks | 40101015 | IAT iShares US Regional Banks | C | 2006 | Alt |
| Regional Banks | 40101015 | KBWR Invesco KBW Regional Banking | D | 2011 | |

**Industry 401020 — Thrifts & Mortgage Finance (Sub-Industry 40102010):** No dedicated liquid ETF.

### Industry Group 4020: Financial Services

**Industry 402030 — Capital Markets**

| Sub-Industry | Code | ETF | Tier | Year | Notes |
|---|---|---|---|---|---|
| Investment Banking & Brokerage | 40203020 | **IAI** iShares US Broker-Dealers & Securities Exchanges | C | 2006 | |
| Capital Markets (broad) | 40203010–30 | KCE SPDR S&P Capital Markets | D | 2005 | |

**Industry 402040 — Mortgage REITs (Sub-Industry 40204010)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **REM** iShares Mortgage Real Estate | C | 2007 | Most liquid mortgage REIT ETF |
| MORT VanEck Mortgage REIT Income | D | 2011 | |

**Cross-sector: FinTech (bridges Financials + IT)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **IPAY** ETFMG Prime Mobile Payments | D | 2015 | Payment processors — GICS 45101020 (Data Processing) |
| FINX Global X FinTech | D | 2016 | Broader fintech |
| BIZD VanEck BDC Income | D | 2013 | Business Development Companies |

**Industry 402020 — Consumer Finance (Sub-Industry 40202010):** No dedicated liquid ETF.

### Industry Group 4030: Insurance

**Industry 403010 — Insurance**

| Sub-Industry | Code | ETF | Tier | Year | Notes |
|---|---|---|---|---|---|
| Broad Insurance | all | **KIE** SPDR S&P Insurance | C | 2005 | Most liquid insurance ETF, equal-weighted |
| Broad Insurance | all | IAK iShares US Insurance | C | 2006 | Alt |
| P&C Insurance | 40301040 | KBWP Invesco KBW Property & Casualty Insurance | D | 2010 | Narrowest dedicated P&C ETF |

No dedicated ETF for Insurance Brokers, Life & Health, or Reinsurance alone.

---

## Sector 45 — Information Technology

Sector ETFs: **XLK** [A, 1998], VGT [A, 2004], IYW [B, 2000], FTEC [B, 2013]

### Industry Group 4510: Software & Services

**Industry 451020 — Software (Sub-Industries 45102010/45102020 Application + Systems Software)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **IGV** iShares Expanded Tech-Software Sector | B | 2001 | Oldest & most liquid software ETF |
| PSJ Invesco Dynamic Software | D | 2005 | Equal-weighted |

**Industry 451010 — IT Services (Sub-Industry 45101020 Data Processing & Outsourced Services)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **IPAY** ETFMG Prime Mobile Payments | D | 2015 | Payment processing focus |
| FINX Global X FinTech | D | 2016 | Broader IT services/fintech |

**Cybersecurity (Sub-Industry 45102020 Systems Software)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **CIBR** First Trust Nasdaq Cybersecurity | C | 2015 | Most liquid cybersecurity ETF (~$6B AUM) |
| HACK ETFMG Prime Cyber Security | C | 2014 | Oldest dedicated cybersecurity ETF |

**Cloud Computing (Internet Services & Infrastructure)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **SKYY** First Trust Cloud Computing | C | 2011 | Oldest cloud ETF |
| CLOU Global X Cloud Computing | D | 2019 | |
| WCLD WisdomTree Cloud Computing | D | 2019 | |

**Robotics & AI (bridges Industrials + IT, GICS 20106020 + 45102010)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **ROBO** ROBO Global Robotics & Automation | C | 2013 | Oldest robotics ETF |
| BOTZ Global X Robotics & AI | C | 2016 | |

### Industry Group 4520: Technology Hardware & Equipment

**Industry 452010 — Communications Equipment (Sub-Industry 45201020):** No dedicated liquid ETF.

**Industry 452020 — Technology Hardware, Storage & Peripherals (Sub-Industry 45202010):** No dedicated liquid ETF. This is essentially Apple + a handful of others; XLK/VGT capture it at sector level.

**Industry 452030 — Electronic Equipment, Instruments & Components:** No dedicated liquid ETF for any sub-industry.

### Industry Group 4530: Semiconductors & Semiconductor Equipment

**Industry 453010 — Semiconductors (Sub-Industry 45301020)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **SOXX** iShares Semiconductor | A | 2001 | Oldest & largest semiconductor ETF; liquid options chain |
| **SMH** VanEck Semiconductor | A | 2000/2011† | Equally liquid; market-cap weighted — extremely NVDA-heavy in recent years |
| SOXQ Invesco PHLX Semiconductor | C | 2021 | Lower expense ratio |

† SMH originated as a HOLDRS Trust (2000), restructured as current VanEck ETF in 2011.

No pure semiconductor equipment ETF — SOXX/SMH include equipment names (AMAT, LRCX, ASML) alongside chip designers and manufacturers.

---

## Sector 50 — Communication Services

Sector ETFs: **XLC** [B, 2018], VOX [C, 2004†], FCOM [C, 2013]

† The 2018 GICS restructure moved Alphabet, Meta, Netflix from Tech/Consumer Disc into Communication Services. XLC was launched for this; VOX was restructured. IYZ (telecom-only, pre-2018 mandate) remains a valid sub-sector ETF.

### Industry Group 5010: Telecommunication Services

**Industry 501010 — Diversified Telecommunication Services (Sub-Industry 50101020 Integrated Telecom)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **IYZ** iShares US Telecommunications | C | 2000 | Oldest telecom ETF; tracks legacy telecom-only definition |

**Industry 501020 — Wireless Telecommunication Services (Sub-Industry 50102010):** No dedicated liquid US ETF.

### Industry Group 5020: Media & Entertainment

**Industry 502020 — Entertainment, Sub-Industry 50202020 Interactive Home Entertainment (Gaming)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **ESPO** VanEck Video Gaming & Esports | D | 2018 | Global gaming exposure |
| HERO Global X Video Games & Esports | D | 2019 | |
| BETZ Roundhill Sports Betting & iGaming | D | 2019 | Skewed toward online gambling and sports betting |

**Industry 502010 — Media (Sub-Industries 50201010–50201040 Advertising, Broadcasting, Cable, Publishing)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **PBS** Invesco Dynamic Media | E | 2005 | Oldest media ETF; small AUM — verify spreads |

**Industry 502030 — Interactive Media & Services (Sub-Industry 50203010):** No dedicated ETF. This sub-industry is dominated by Alphabet and Meta, which make up 35–45% of XLC/VOX by weight.

---

## Sector 55 — Utilities

Sector ETFs: **XLU** [A, 1998], VPU [B, 2004], IDU [C, 2000]

### Industry Group 5510: Utilities

**Industry 551040 — Water Utilities (Sub-Industry 55104010)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **PHO** Invesco Water Resources | C | 2005 | Oldest & most liquid water ETF; spans utilities + treatment infrastructure |
| FIW First Trust Water | D | 2007 | Pure water utilities + purification |

**Industry 551050 — Renewable Electricity (Sub-Industry 55105020)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **ICLN** iShares Global Clean Energy | B | 2008 | Most liquid clean energy ETF (wind + solar + utilities) |
| **QCLN** First Trust NASDAQ Clean Edge Green Energy | C | 2007 | Oldest US clean energy ETF |
| TAN Invesco Solar | C | 2008 | Pure solar (global) |
| FAN First Trust Global Wind Energy | D | 2008 | Pure wind (global) |

**Industries 551010–551030 — Electric Utilities; Gas Utilities; Multi-Utilities; Independent Power Producers:** No dedicated liquid ETF for any individual sub-industry. XLU is the proxy, and despite technically covering the entire sector, its holdings are ~65–70% electric utilities in practice.

---

## Sector 60 — Real Estate

Sector ETFs: **VNQ** [A, 2004], XLRE [B, 2015], IYR [B, 2000], RWR [C, 2001], SCHH [B, 2011]

### Industry Group 6010: Equity REITs

**Industry 601025 — Industrial REITs (Sub-Industry 60102510)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **INDS** Pacer Benchmark Industrial Real Estate SCTR | D | 2018 | Only dedicated industrial REIT ETF |

**Industry 601060 — Residential REITs (Sub-Industries 60106010 Multi-Family + 60106020 Single-Family)**

| ETF | Tier | Year | Notes |
|---|---|---|---|
| **REZ** iShares Residential & Multisector Real Estate | D | 2007 | Residential + healthcare REITs |

**Industry 601080 — Specialized REITs**

| Sub-Industry | Code | ETF | Tier | Year | Notes |
|---|---|---|---|---|---|
| Data Center REITs | 60108050 | **SRVR** Pacer Data & Infrastructure Real Estate SCTR | D | 2018 | Data centres + cell towers |
| Timber REITs | 60108040 | **WOOD** iShares Global Timber & Forestry | D | 2008 | Dual-listed: also covers Materials forestry (15105010) |

**Industries with no dedicated liquid ETF:**
- 601010 Diversified REITs → VNQ proxy
- 601030 Hotel & Resort REITs → VNQ proxy
- 601040 Office REITs → VNQ proxy
- 601050 Health Care REITs → VNQ proxy
- 601070 Retail REITs → VNQ proxy

### Industry Group 6020: Real Estate Management & Development

Sub-Industries 60201010–60201040 (Diversified RE Activities, Operating Cos, Development, Services): No dedicated ETF.

---

## Coverage Summary

| GICS Level | Total | ETF Coverage | Coverage % |
|---|---|---|---|
| Sectors | 11 | 11 | 100% |
| Industry Groups | 25 | ~20 | ~80% |
| Industries | 74 | ~28 | ~38% |
| Sub-Industries | 163 | ~35 | ~21% |

---

## Structural Gaps — Notable Missing ETFs

These GICS nodes have no clean dedicated ETF; you must use the sector-level fund as a noisy proxy:

- **Chemicals** (all sub-industries: commodity, diversified, specialty, industrial gases, fertilizers) → XLB
- **Containers & Packaging** → XLB
- **Coal** → KOL liquidated 2020, no replacement
- **Building Products, Construction & Farm Machinery, Electrical Equipment** → XLI or PAVE
- **Professional Services, Commercial Printing, Security Services** → XLI
- **Marine Transportation, Rail, Trucking, Transport Infrastructure** → IYT
- **Automobile Components, Motorcycle Manufacturers** → XLY
- **Apparel, Footwear, Luxury Goods, Textiles** → XLY
- **Restaurants** → PEJ is partial; no pure play
- **Beverages (Brewers, Distillers, Soft Drinks individually), Tobacco** → XLP
- **Drug Retail, Food Retail, Food Distributors** → XLP
- **Household Products, Personal Care** → XLP
- **Life Sciences Tools & Services** → XLV
- **Health Care Technology / Digital Health** → XLV (small thematic ETFs exist but AUM is sub-$100M)
- **Communications Equipment, Technology Hardware, Electronic Components, Electronic Mfg Services** → XLK
- **Semiconductor Equipment (standalone)** → SOXX/SMH include it but it's not isolated
- **Wireless Telecom** → XLC
- **Interactive Media (Alphabet/Meta sub-industry)** → XLC, ~40% of the weight
- **Electric Utilities, Gas Utilities, Multi-Utilities individually** → XLU
- **Office REITs, Hotel REITs, Retail REITs, Health Care REITs individually** → VNQ

---

## Practical Notes

**Spread tax reality by tier.** For a $50k account rebalancing frequently: [A]/[B] ETFs have sub-$0.01 spreads and you can ignore impact. [C]/[D] ETFs run $0.01–$0.03 and are fine unless you're rebalancing daily. [E] ETFs hit $0.03–$0.07 — you're paying 2–5bps per side, still manageable if turnover is weekly/monthly. [F] ETFs (EVX, CRAK, PBS) can gap to $0.10+ intraday — time your entries to liquid market hours (10:00–15:30 ET) and use limit orders.

**Biotech dual-use warning.** XBI and IBB are both "Biotechnology" but represent completely different factor exposures. XBI is equal-weighted small/mid biotech with binary FDA-event risk and 50%+ annual drawdowns in stress. IBB is cap-weighted mega-cap biotech acting more like a defensive growth ETF. Running both simultaneously is intentional only if you want separate large/small biotech legs.

**Communication Services post-2018 composition.** XLC is highly concentrated — the top 5 names (GOOGL, META, NFLX, ATVI, DIS historically) can represent 40%+ of the fund. VOX is marginally more diversified. IYZ gives you the old telecom-only exposure (AT&T, Verizon, T-Mobile) without the internet giants.

**MLP tax structures.** AMLP uses a C-corp wrapper which creates a tax drag of roughly 1–2% annually relative to owning the MLPs directly. MLPA and MLPX avoid this. For systematic strategies with frequent rebalancing, the structure matters less than for buy-and-hold; just model it correctly in your cost assumptions.

**Semiconductor concentration.** SMH has had >20% in NVDA at peak. SOXX is more diversified by index construction (capped weighting). If you're using both in a multi-strategy portfolio, you're doubling semiconductor factor exposure — be explicit about this in your factor risk model.

**Verify before going live.** ETF tickers and AUM shift. Any fund with [E] or [F] tier should be checked on ETF.com or the issuer's site for current AUM, expense ratio, and whether the fund is still in operation before including in any live strategy.

```python
# ============================================================
# GICS ETF UNIVERSE
# Primary ETF only per sub-sector (no alternatives)
# Tiers: [A] >$10B  [B] $3-10B  [C] $1-3B  [D] $300M-1B  [E] $100-300M  [F] <$100M
# Spread notes: [E] use limits; [F] trade 10:00-15:30 ET only, verify AUM before live
# ============================================================

# --- GICS Level 1: Sectors ---
sector_ETFs = [
    "XLE",    # Energy                        [A] 1998
    "XLB",    # Materials                     [B] 1998
    "XLI",    # Industrials                   [A] 1998
    "XLY",    # Consumer Discretionary        [A] 1998
    "XLP",    # Consumer Staples              [A] 1998
    "XLV",    # Health Care                   [A] 1998
    "XLF",    # Financials                    [A] 1998
    "XLK",    # Information Technology        [A] 1998
    "XLC",    # Communication Services        [B] 2018
    "XLU",    # Utilities                     [A] 1998
    "VNQ",    # Real Estate                   [A] 2004
]

# --- Energy Sub-Sector (GICS 10) ---
energy_sub_ETFs = [
    "OIH",    # Oil Equipment & Services      [C] 2001  Sub-Industry 10101020
    "XOP",    # Oil & Gas E&P                 [B] 2006  Sub-Industry 10102020
    "CRAK",   # Oil Refiners                  [F] 2015  Sub-Industry 10102030 — watch spreads
    "AMLP",   # MLPs / Pipelines              [B] 2010  Sub-Industry 10102040
]

# --- Materials Sub-Sector (GICS 15) ---
# Note: GDX/GDXJ and SIL/SILJ are NOT alternatives — senior vs junior miners
# are genuinely different size-factor exposures within the same sub-industry
materials_sub_ETFs = [
    "XME",    # Metals & Mining broad         [C] 2006  Industry 151040
    "COPX",   # Copper Miners                 [C] 2010  Sub-Industry 15104025
    "GDX",    # Gold Miners (senior)          [B] 2006  Sub-Industry 15104030
    "GDXJ",   # Gold Miners (junior)          [B] 2009  Sub-Industry 15104030
    "SIL",    # Silver Miners (senior)        [D] 2010  Sub-Industry 15104045
    "SILJ",   # Silver Miners (junior)        [D] 2012  Sub-Industry 15104045
    "SLX",    # Steel                         [D] 2006  Sub-Industry 15104050
    "WOOD",   # Timber & Forestry             [D] 2008  Industry 151050 + REIT 60108040
]

# --- Industrials Sub-Sector (GICS 20) ---
industrials_sub_ETFs = [
    "ITA",    # Aerospace & Defense           [B] 2006  Industry 201010
    "PAVE",   # Infrastructure / Const.       [B] 2017  Industry 201030
    "IYT",    # Transportation broad          [C] 2003  Industry Group 2030
    "JETS",   # Airlines                      [C] 2015  Sub-Industry 20302010
    "EVX",    # Environmental Services        [F] 2006  Sub-Industry 20201050 — watch spreads
]

# --- Consumer Discretionary Sub-Sector (GICS 25) ---
consumer_disc_sub_ETFs = [
    "CARZ",   # Automobile Manufacturers      [E] 2011  Sub-Industry 25102010
    "ITB",    # Homebuilders                  [B] 2006  Sub-Industry 25201030
    "XRT",    # Retail broad                  [B] 2006  Industry Groups 2540/2550
    "IBUY",   # Online / E-Commerce Retail    [D] 2016  Sub-Industry 25402010
    "PEJ",    # Hotels, Leisure & Restaurants [D] 2005  Industry 253010
    "BETZ",   # Casinos & Gaming              [D] 2019  Sub-Industry 25301010
]

# --- Consumer Staples Sub-Sector (GICS 30) ---
consumer_staples_sub_ETFs = [
    "PBJ",    # Packaged Foods & Beverages    [D] 2005  Industry 302020
    "MOO",    # Agribusiness                  [C] 2007  Sub-Industry 30202010
]

# --- Health Care Sub-Sector (GICS 35) ---
# WARNING: IBB and XBI are NOT interchangeable — distinct factor exposures:
#   IBB = cap-weighted mega-cap biotech (defensive growth, behaves like XLV)
#   XBI = equal-weighted small/mid biotech (binary FDA-event risk, 50%+ drawdown possible)
healthcare_sub_ETFs = [
    "IHI",    # Medical Devices               [B] 2006  Sub-Industry 35101010
    "IHF",    # HC Providers & Services       [C] 2006  Industry 351020
    "IBB",    # Biotech large-cap             [A] 2001  Sub-Industry 35201010
    "XBI",    # Biotech small/mid-cap         [B] 2006  Sub-Industry 35201010 — different exposure
    "IHE",    # Pharmaceuticals               [D] 2006  Sub-Industry 35202010
]

# --- Financials Sub-Sector (GICS 40) ---
financials_sub_ETFs = [
    "KRE",    # Regional Banks                [B] 2006  Sub-Industry 40101015
    "KBE",    # Diversified Banks broad       [C] 2005  Sub-Industry 40101010
    "REM",    # Mortgage REITs                [C] 2007  Sub-Industry 40204010
    "IAI",    # Capital Markets / Brokers     [C] 2006  Industry 402030
    "KIE",    # Insurance broad               [C] 2005  Industry 403010
    "IPAY",   # Mobile Payments / FinTech     [D] 2015  Sub-Industry 45101020 (bridges IT/Fins)
    "BIZD",   # Business Development Cos.     [D] 2013  BDCs (credit-adjacent Financials)
]

# --- Information Technology Sub-Sector (GICS 45) ---
technology_sub_ETFs = [
    "SOXX",   # Semiconductors                [A] 2001  Sub-Industry 45301020
    "IGV",    # Software (App + Systems)      [B] 2001  Sub-Industries 45102010/20
    "CIBR",   # Cybersecurity                 [C] 2015  Sub-Industry 45102020
    "SKYY",   # Cloud Computing               [C] 2011  Internet Infrastructure
    "ROBO",   # Robotics & AI                 [C] 2013  GICS 20106020 + 45102010
]

# --- Communication Services Sub-Sector (GICS 50) ---
comm_services_sub_ETFs = [
    "IYZ",    # Telecom (legacy: T/VZ/TMUS)   [C] 2000  Industry 501010
    "ESPO",   # Video Gaming & Esports         [D] 2018  Sub-Industry 50202020
    "PBS",    # Media (Advertising/Cable)      [E] 2005  Industry 502010 — watch spreads
]

# --- Utilities Sub-Sector (GICS 55) ---
# Note: TAN and FAN are both Sub-Industry 55105020 (Renewable Electricity)
# but target distinct generation technologies — not alternatives
utilities_sub_ETFs = [
    "PHO",    # Water Utilities               [C] 2005  Sub-Industry 55104010
    "ICLN",   # Clean / Renewable Energy      [B] 2008  Sub-Industry 55105020 (broad)
    "TAN",    # Solar                         [C] 2008  Sub-Industry 55105020 (solar)
    "FAN",    # Wind                          [D] 2008  Sub-Industry 55105020 (wind)
]

# --- Real Estate Sub-Sector (GICS 60) ---
# WOOD (Timber REITs, Sub-Industry 60108040) already captured in materials_sub_ETFs
real_estate_sub_ETFs = [
    "INDS",   # Industrial REITs              [D] 2018  Sub-Industry 60102510
    "REZ",    # Residential REITs             [D] 2007  Sub-Industry 60106010/20
    "SRVR",   # Data Centre & Tower REITs     [D] 2018  Sub-Industry 60108050
]

# ============================================================
# Combined Universes
# ============================================================

gics_sub_ETFs = list(set(
    energy_sub_ETFs +
    materials_sub_ETFs +
    industrials_sub_ETFs +
    consumer_disc_sub_ETFs +
    consumer_staples_sub_ETFs +
    healthcare_sub_ETFs +
    financials_sub_ETFs +
    technology_sub_ETFs +
    comm_services_sub_ETFs +
    utilities_sub_ETFs +
    real_estate_sub_ETFs
))  # 52 unique sub-sector ETFs

gics_all_ETFs = list(set(sector_ETFs + gics_sub_ETFs))  # 63 total
```

---

**A few things to be aware of in code:**

`list(set(...))` randomises order. If your downstream code cares about ordering (e.g., aligning to a factor matrix, building a correlation heatmap), sort explicitly: `sorted(list(set(...)))`.

**[F] tier names to gate in live code** — consider a separate `illiquid_ETFs = ["CRAK", "EVX"]` set that you filter out of any strategy that rebalances more frequently than weekly, or route through a spread-check before ordering.

**The IBB/XBI split** matters in any momentum or cross-sectional ranking strategy. If both are in your universe and both are in the top decile of momentum, you're doubling healthcare biotech factor exposure without realising it. Worth building explicit sector-exposure bookkeeping from the start.

Listen up. As a retail systematic trader working with a low budget, your biggest enemies aren't just bad signals—they are **slippage, bid-ask spreads, and expense ratios**. 

When you look at a map like Finviz, you’ll see dozens of ETFs, but a seasoned quant knows that 80% of them are untradeable for a fast-moving or low-budget systematic strategy because they lack liquidity. You need US-listed ETFs with high Average Daily Volume (ADV) and tight spreads so your algorithms can enter and exit without getting chopped to pieces.

For country and regional rotation strategies, the **iShares MSCI (E-series)** and **Vanguard** families are the gold standard. Here is the definitive list of highly liquid country and regional ETFs for all major economies, categorized just like the Finviz map.

### 1. Broad Regional (The Macro Bets)
If your model trades macroeconomic regimes, these are your most liquid instruments. Vanguard is cheaper to hold (lower expense ratios), but iShares often have better options chains and intraday liquidity.
*   **World (ex-US):** `VEA` (Vanguard Developed Markets) or `EFA` (iShares MSCI EAFE)
*   **Emerging Markets:** `VWO` (Vanguard Emerging Markets) or `EEM` (iShares MSCI Emerging)
*   **Europe Broad:** `VGK` (Vanguard European) or `EZU` (iShares Eurozone)
*   **Asia Pacific Broad:** `VPL` (Vanguard Pacific) or `AAXJ` (iShares Asia ex-Japan)

### 2. North & South America
*   **United States:** `SPY` (S&P 500), `QQQ` (Nasdaq 100), `IWM` (Russell 2000)
*   **Canada:** `EWC` (iShares MSCI Canada)
*   **Mexico:** `EWW` (iShares MSCI Mexico)
*   **Brazil:** `EWZ` (iShares MSCI Brazil) — *Quant Note: EWZ is highly volatile and highly liquid; it's a favorite for mean-reversion systems.*
*   **Latin America (Broad):** `ILF` (iShares Latin America 40)

### 3. Europe (Developed)
*   **United Kingdom:** `EWU` (iShares MSCI United Kingdom)
*   **Germany:** `EWG` (iShares MSCI Germany)
*   **France:** `EWQ` (iShares MSCI France)
*   **Switzerland:** `EWL` (iShares MSCI Switzerland)
*   **Italy:** `EWI` (iShares MSCI Italy)
*   **Spain:** `EWP` (iShares MSCI Spain)

### 4. Asia-Pacific
*   **Japan:** `EWJ` (iShares MSCI Japan) — *Massive liquidity, highly institutionalized.*
*   **China (Large Cap):** `FXI` (iShares China Large-Cap)
*   **China (Broad/Tech):** `MCHI` (iShares MSCI China) or `KWEB` (KraneShares CSI China Internet)
*   **India:** `INDA` (iShares MSCI India) or `EPI` (WisdomTree India Earnings)
*   **South Korea:** `EWY` (iShares MSCI South Korea)
*   **Taiwan:** `EWT` (iShares MSCI Taiwan) — *Essentially a semi-conductor play (TSMC dominates this index).*
*   **Australia:** `EWA` (iShares MSCI Australia)

### 5. Middle East & Africa
*   **Saudi Arabia:** `KSA` (iShares MSCI Saudi Arabia)
*   **South Africa:** `EZA` (iShares MSCI South Africa)

---

### 🧠 The Quant's Survival Guide for Trading These (Read Carefully)

Since you are running automated/systematic strategies on a low budget, I'm going to save you thousands of dollars right now:

1.  **Beware of "Stale Pricing" (Time Zone Disconnect):** 
    When you are trading `EWJ` (Japan) or `EWG` (Germany) at 2:00 PM EST, **their underlying stock markets are closed**. The ETF price you see is being maintained by US market makers using correlated US assets (like S&P futures or currency pairs) to guess where the foreign market *will* open. If your model uses intraday standard deviation or breakout logic on foreign ETFs during US hours, your signals are mostly reacting to US market noise, not local country news. 
2.  **Trade at the European Close:**
    If your system trades European ETFs (`EWG`, `EWU`, `EWQ`), execute your trades between 10:30 AM and 11:30 AM EST. The European markets close at 11:30 AM EST. After that, liquidity dries up and spreads widen because the European market makers log off.
3.  **Spread-to-Price Ratio:**
    Because you have a small account, you might be tempted to buy cheap ETFs. Don't look at the share price; look at the **spread as a percentage of the price**. A 2-cent spread on a $10 ETF is a 0.2% loss the second you enter. A 2-cent spread on a $50 ETF is a 0.04% loss. Stick to the tickers above; they usually trade at a 1-cent spread.
4.  **Currency Drag (Unhedged vs. Hedged):**
    All the `E-series` ETFs above are **unhedged**. That means if you buy `EWG` (Germany), you are long German stocks *and* long the Euro. If the Euro drops against the Dollar, your ETF goes down even if German stocks went up. If your model is specifically trying to trade equity momentum, you might want to look at currency-hedged ETFs (like `HEWJ` for Japan or `HEWG` for Germany), though they are less liquid.
5.  **Avoid Frontier Markets:**
    You'll see ETFs for places like Nigeria, Vietnam, or Colombia on Finviz. **Ignore them.** Their bid-ask spreads are wide enough to drive a truck through. As a retail systematic trader, if an ETF doesn't trade at least 1,000,000 shares a day, exclude it from your universe.

Here is the complete, updated Python code universe, now including the highly liquid major **Leveraged (Bull)** and **Inverse/Leveraged Inverse (Bear)** ETFs. 

I have limited these new lists to the heavyweights—the ones that trade millions of shares a day. In systematic trading, you should avoid niche leveraged ETFs (like 3x Retail or 3x Homebuilders) because their spreads will destroy your edge. The big index and semiconductor leveraged funds, however, have penny spreads and deep liquidity.

```python
# 1. Broad Market / Size ETFs
size_ETFs = [
    "SPY", "IVV", "VOO", # S&P 500
    "QQQ",               # Nasdaq 100
    "DIA",               # Dow Jones
    "MDY", "IJH", "VO",  # Mid-Cap
    "IWM", "IJR", "VB"   # Small-Cap
]

# 2. Factor & Style ETFs
factor_ETFs = [
    "VTV", "IVE", "VLUE", # Value
    "VUG", "IVW",         # Growth
    "MTUM",               # Momentum
    "QUAL",               # Quality
    "USMV", "SPLV",       # Low Volatility
    "SCHD", "VIG", "VYM"  # Dividend Growth/Yield
]

# 3. Fixed Income / Bond ETFs
bond_ETFs = [
    "BIL", "SHV", "SHY",  # Ultra-Short / Short Treasury
    "IEI", "IEF",         # Intermediate Treasury
    "TLT", "EDV",         # Long-Term Treasury
    "AGG", "BND",         # Total Aggregate Bond
    "LQD", "VCIT", "VCSH",# Inv Grade Corporate
    "HYG", "JNK",         # High Yield Corporate
    "MBB",                # Mortgage-Backed
    "MUB"                 # Municipal
]

# 4. US Sector ETFs
sector_ETFs = [
    "XLE", "XLF", "XLK", "XLV", "XLI", "XLY", 
    "XLP", "XLU", "XLB", "XLRE", "XLC", "XBI"
]

# 5. Broad International / Region ETFs
region_ETFs = [
    "VEA", "EFA",         # Developed ex-US
    "VWO", "EEM",         # Emerging Markets
    "VXUS", "IXUS"        # Total Int ex-US
]

# 6. Sub-Region ETFs
subregion_ETFs = [
    "VGK", "EZU",         # Europe
    "VPL", "AAXJ",        # Asia Pacific
    "ILF"                 # Latin America
]

# 7. Single Country ETFs
country_ETFs = [
    "EWC", "EWU", "EWG", "EWQ", "EWL", "EWI", "EWP", "EWD", 
    "EWJ", "EWA", "EWY", "EWT", "INDA", "FXI", "MCHI", 
    "EWZ", "EWW", "EZA", "KSA"
]

# 8. Physical Commodities ETFs
commodity_ETFs = [
    "GLD", "IAU", "SLV", "USO", "BNO", "UNG", "DBC", "PDBC", "GSG"
]

# 9. Volatility / VIX ETFs
volatility_ETFs = [
    "VXX", "VIXY", "SVXY" 
]

# 10. Leveraged ETFs (Bull - 2x and 3x)
leveraged_ETFs = [
    "UPRO", "SSO",        # 3x and 2x S&P 500
    "TQQQ", "QLD",        # 3x and 2x Nasdaq 100
    "UDOW",               # 3x Dow Jones
    "TNA",  "UWM",        # 3x and 2x Russell 2000
    "SOXL",               # 3x Semiconductors
    "TECL",               # 3x Technology Sector
    "FAS",                # 3x Financials
    "TMF",                # 3x Long-Term Treasury (TLT equivalent)
    "UCO"                 # 2x Crude Oil
]

# 11. Inverse & Leveraged Inverse ETFs (Bear - -1x, -2x, and -3x)
inverse_ETFs = [
    "SPXU", "SDS", "SH",  # -3x, -2x, and -1x S&P 500
    "SQQQ", "QID", "PSQ", # -3x, -2x, and -1x Nasdaq 100
    "SDOW",               # -3x Dow Jones
    "TZA",  "RWM",        # -3x and -1x Russell 2000
    "SOXS",               # -3x Semiconductors
    "TECS",               # -3x Technology Sector
    "FAZ",                # -3x Financials
    "TMV"                 # -3x Long-Term Treasury
]

# Combine all into a master universe list, removing duplicates
all_ETFs = list(set(
    size_ETFs + factor_ETFs + bond_ETFs + sector_ETFs + 
    region_ETFs + subregion_ETFs + country_ETFs + 
    commodity_ETFs + volatility_ETFs + 
    leveraged_ETFs + inverse_ETFs
))

# Print length check
print(f"Total Unique Highly Liquid ETFs in Universe: {len(all_ETFs)}")
```

### ⚠️ The Quant's Warning on Leveraged & Inverse ETFs

Since you are coding a systematic strategy, handling `TQQQ`, `SQQQ`, `SOXL`, or `SOXS` requires extreme caution. If you naively treat these like normal stocks in your backtester, your results will be dangerously inaccurate. Here is what you must account for in your code:

1. **The "Volatility Drag" (Beta Slippage):**
   These ETFs are designed to track the *daily* percentage change of the index, not the long-term change. 
   * *The Math:* If the QQQ goes down 10% on Monday, `TQQQ` drops 30%. To get back to even, the QQQ needs to go up 11.1%. But `TQQQ` now needs to go up **42.8%** just to break even! 
   * *The Code Implication:* Do not hold 2x/3x ETFs in a systematic swing or trend-following strategy that keeps positions open for months, especially during choppy, sideways markets. The math practically guarantees the ETF will decay over time. Use them for short-term mean-reversion (1-5 days) or pure intraday momentum.
2. **Reverse Splits Will Break Bad Data:**
   Because of volatility drag, inverse ETFs (like `SQQQ` and `SOXS`) inevitably grind toward zero over the years. To keep the share price tradeable, the fund managers frequently execute reverse splits (e.g., 1-for-5 or 1-for-10). If your data provider doesn't correctly handle historical split adjustments, your backtest will see a "jump" from $5 to $25 overnight and register a massive fake profit or loss. Always sanity-check inverse ETF data.
3. **Calculate Signals on the Underlying, NOT the Leveraged ETF:**
   If you want to trade `TQQQ` using a moving average crossover or an RSI signal, **do not calculate the indicator on TQQQ's price data**. Calculate your RSI and Moving Averages using `QQQ` (the underlying asset). Generate the trade signal from `QQQ`, and then execute the trade in `TQQQ`. Why? Because the daily compounding of the leveraged ETF alters the historical volatility profile, which warps technical indicators and statistical z-scores.
4. **Hard-to-Borrow (HTB) Costs:**
   If your strategy attempts to short inverse ETFs (e.g., shorting `SQQQ` to capture its structural decay—a common quant trade), be aware that these are often Hard-to-Borrow. Retail brokers will charge you hefty annualized borrow fees. Ensure your backtester can simulate these borrow costs, or your 15% annualized backtested gain might actually be a net loss after fees.

For a systematic trader, I would split this into **three universes**:

1. **Country / Region ETFs** (macro rotation, intermarket analysis)
2. **Sector / Industry ETFs** (GICS-style relative strength)
3. **Inverse ETFs** (bear-market and short exposure)

The reality is that the ETF industry does **not** offer inverse ETFs for all 163 GICS sub-industries. Coverage becomes sparse below the sector and industry-group level. The deepest inverse coverage is at:

* Broad market indexes
* Sectors
* Semiconductors
* Biotech
* Real estate
* Regional equity indexes
* Some thematic areas

---

# 1. Country / Region ETFs (Major Economies)

For systematic trading, I would prioritize **iShares country ETFs** because they are old, liquid, and widely used.

| Country      | ETF  |
| ------------ | ---- |
| USA          | SPY  |
| Canada       | EWC  |
| Mexico       | EWW  |
| Brazil       | EWZ  |
| Argentina    | ARGT |
| Chile        | ECH  |
| Peru         | EPU  |
| UK           | EWU  |
| Germany      | EWG  |
| France       | EWQ  |
| Switzerland  | EWL  |
| Italy        | EWI  |
| Spain        | EWP  |
| Netherlands  | EWN  |
| Sweden       | EWD  |
| Norway       | ENOR |
| Denmark      | EDEN |
| Finland      | EFNL |
| Japan        | EWJ  |
| China        | MCHI |
| Hong Kong    | EWH  |
| Taiwan       | EWT  |
| South Korea  | EWY  |
| India        | INDA |
| Singapore    | EWS  |
| Indonesia    | EIDO |
| Thailand     | THD  |
| Malaysia     | EWM  |
| Vietnam      | VNM  |
| Australia    | EWA  |
| New Zealand  | ENZL |
| South Africa | EZA  |
| Saudi Arabia | KSA  |
| UAE          | UAE  |
| Turkey       | TUR  |
| Israel       | EIS  |

---

# 2. Regional ETFs

These are extremely useful for regime and relative-strength models.

| Region            | ETF  |
| ----------------- | ---- |
| Europe            | VGK  |
| Developed Markets | EFA  |
| Emerging Markets  | EEM  |
| Frontier Markets  | FM   |
| Asia ex Japan     | AAXJ |
| Pacific           | VPL  |
| Latin America     | ILF  |
| BRIC              | BKF  |
| ASEAN             | ASEA |
| GCC/Middle East   | GULF |

---

# 3. Major Inverse Market ETFs

These are the core bear-market instruments.

| Exposure             | ETF  |
| -------------------- | ---- |
| S&P 500              | SH   |
| S&P 500 2x Bear      | SDS  |
| S&P 500 3x Bear      | SPXU |
| Nasdaq 100           | PSQ  |
| Nasdaq 100 2x Bear   | QID  |
| Nasdaq 100 3x Bear   | SQQQ |
| Russell 2000         | RWM  |
| Russell 2000 2x Bear | TWM  |
| Russell 2000 3x Bear | TZA  |
| Dow Jones            | DOG  |
| Dow Jones 2x Bear    | DXD  |
| Dow Jones 3x Bear    | SDOW |
| Mid Cap 400          | MYY  |
| Mid Cap 400 2x Bear  | MVV  |
| Mid Cap 400 3x Bear  | SMDD |

---

# 4. Inverse GICS Sector ETFs

This is where most systematic traders stop.

| GICS Sector            | Inverse ETF                 |
| ---------------------- | --------------------------- |
| Technology             | REW                         |
| Financials             | SEF                         |
| Industrials            | SIJ                         |
| Consumer Discretionary | SCC                         |
| Consumer Staples       | SZK                         |
| Energy                 | DUG                         |
| Materials              | SMN                         |
| Real Estate            | DRV                         |
| Utilities              | SDP                         |
| Healthcare             | RXD                         |
| Communication Services | no major liquid inverse ETF |

---

# 5. Semiconductor Inverse ETFs

Semiconductors are arguably the most important industry-level ETF group.

| Industry          | ETF  |
| ----------------- | ---- |
| Semiconductors    | SOXS |
| Semiconductors 2x | SSG  |

For many quants, SOXS is effectively the inverse ETF for:

* Semiconductors
* Semiconductor Equipment
* Foundries
* Chip Designers
* AI Hardware

---

# 6. Biotech / Healthcare Inverse ETFs

| Industry              | ETF  |
| --------------------- | ---- |
| Biotechnology         | BIS  |
| Biotechnology 3x Bear | LABD |
| Healthcare            | RXD  |

---

# 7. Real Estate Inverse ETFs

| Industry    | ETF |
| ----------- | --- |
| Real Estate | DRV |
| REITs       | SRS |

---

# 8. Emerging Market & International Inverse ETFs

| Region           | ETF                                           |
| ---------------- | --------------------------------------------- |
| Emerging Markets | EEV                                           |
| China            | YANG                                          |
| China Internet   | SZI                                           |
| Europe           | EPV                                           |
| Europe 3x Bear   | EURL (leveraged long opposite side available) |
| MSCI EAFE        | EFZ                                           |

---

# 9. Commodity Producer Inverse ETFs

| Industry                 | ETF                  |
| ------------------------ | -------------------- |
| Oil & Gas Producers      | DRIP                 |
| Exploration & Production | DRIP                 |
| Gold Miners              | DUST                 |
| Junior Gold Miners       | JDST                 |
| Silver Miners            | ZSL                  |
| Uranium Miners           | no major inverse ETF |

---

# 10. Closest You Can Get to GICS Industry Coverage

For your use case (systematic trading, $50k account), I'd build the hierarchy:

### Level 1 (Countries)

~35 ETFs

### Level 2 (Regions)

~10 ETFs

### Level 3 (Sectors)

11 ETFs

Examples:

* XLK
* XLF
* XLE
* XLI
* XLV
* XLU
* XLB
* XLP
* XLY
* XLRE
* XLC

### Level 4 (Industries)

30–50 ETFs

Examples:

* SMH (Semis)
* KRE (Regional Banks)
* IBB (Biotech)
* XBI (Biotech)
* ITA (Aerospace)
* XRT (Retail)
* IYR (Real Estate)
* PAVE (Infrastructure)
* IGV (Software)
* FDN (Internet)
* HACK (Cybersecurity)
* ARKG (Genomics)
* TAN (Solar)
* URA (Uranium)
* REMX (Rare Earths)

### Level 5 (Inverse Universe)

~25-40 ETFs

Examples:

* SQQQ
* SOXS
* TZA
* SPXU
* SDS
* SH
* PSQ
* REW
* SEF
* SIJ
* SCC
* SMN
* DUG
* DRV
* LABD
* BIS
* DUST
* JDST
* DRIP
* YANG
* EEV
* EFZ

For a low-budget systematic trader, this ETF universe is actually powerful because it lets you model:

* Global country rotation
* Sector rotation
* Industry rotation
* Risk-on / risk-off
* Bear market leadership
* Relative strength
* Intermarket relationships

without needing expensive survivorship-bias-free constituent databases for every stock universe. In practice, many institutional top-down models are built from a universe of only ~100-300 highly liquid ETFs rather than thousands of individual equities.
